sql/select.go

```diff
@@ -378,16 +384,45 @@ func (v *indexInfo) makeConstraints(exprs []parser.Exprs) {
 	startDone := false
 	endDone := false
 
-	for _, colID := range v.index.ColumnIDs {
-		constraint := indexConstraint{}
+	for i := 0; i < len(v.index.ColumnIDs); i++ {
```

Comment:
I advance the value of `i` in the loop below when handling tuple equality expressions. I think it would work correctly to make the change you suggest, but it seemed weird.